apps/browser-extension/src/background/background.ts

```diff
@@ -0,0 +1,45 @@
+import { Settings } from "../utils/settings.ts";
+
+const OPEN_HOARDER_ID = "open-hoarder";
+
+function checkSettingsState(settings: Settings) {
+  if (settings?.address) {
+    registerContextMenu(settings.address);
+  } else {
+    chrome.contextMenus.remove(OPEN_HOARDER_ID);
+  }
+}
+
+/**
+ * Registers a context menu button to open a tab with the currently configured hoarder instance
+ * @param address The address of the configured hoarder instance
+ */
+function registerContextMenu(address: string) {
+  chrome.contextMenus.create({
+    id: OPEN_HOARDER_ID,
+    title: "Open Hoarder",
+    contexts: ["action"],
+  });
+  chrome.contextMenus.onClicked.addListener(createContextClickHandler(address));
+}

```

Comment:
I don't think we should be adding a listener every time the setting changes. We should just always register the listener, and inside the listener (at top level), and inside the listener, we grab the setting object and open the address if it's set.